sdmetrics/reports/utils.py

```diff
@@ -479,6 +479,133 @@ def get_column_pair_plot(real_data, synthetic_data, column_names, metadata):
+            The name of the foreign key column in the child table.
+
+    Returns:
+        pandas.DataFrame
+    """
+    child_counts = child_table[child_foreign_key].value_counts()
+    child_per_parent = child_counts.reindex(parent_table[parent_primary_key], fill_value=0)
+    child_per_parent = child_per_parent.reset_index()
+    child_per_parent.columns = [parent_primary_key, '# children']
+
+    cardinalities = child_per_parent.groupby('# children').size().reset_index(name='# parents')
+
+    return cardinalities.sort_values('# children')
+
+
+def generate_cardinality_plot(data, parent_primary_key, child_foreign_key):
+    """Generate a plot of the cardinality of the parent-child relationship.
+
+    Args:
+        data (pandas.DataFrame):
+            The cardinality data.
+        parent_primary_key (string):
+            The name of the primary key column in the parent table.
+        child_foreign_key (string):
+            The name of the foreign key column in the child table.
+
+    Returns:
+        plotly.graph_objects._figure.Figure
+    """
+    fig = px.histogram(data,
+                       x='# children',
+                       y='# parents',
+                       color='data',
+                       barmode='group',
+                       color_discrete_sequence=[DATACEBO_DARK, DATACEBO_LIGHT],
+                       pattern_shape='data',
+                       pattern_shape_sequence=['', '/'],
+                       nbins=max(data['# children']) - min(data['# children']) + 1,
+                       histnorm='probability density')
+
+    for name in ['Real', 'Synthetic']:
+        fig.update_traces(
+            hovertemplate=f'<b>{name}</b><br>Frequency: %{{y}}<extra></extra>',
+            selector={'name': name}
+        )
+
+    title = (
+        f"Relationship (child foreign key='{child_foreign_key}' and parent "
+        f"primary key='{parent_primary_key}')"
+    )
+    fig.update_layout(
+        title=title,
+        xaxis_title='# of Children (per Parent)',
+        yaxis_title='Frequency',
+        plot_bgcolor=BACKGROUND_COLOR
+    )
+    return fig
+
+
+def get_cardinality_plot(real_data, synthetic_data, child_table_name, parent_table_name,
+                         child_foreign_key, metadata):
+    """Return a plot of the cardinality of the parent-child relationship.
+
+    Args:
+        real_data (dict):
+            The real data.
+        synthetic_data (dict):
+            The synthetic data.
+        child_table_name (string):
+            The name of the child table.
+        parent_table_name (string):
+            The name of the parent table.
+        child_foreign_key (string):
+            The name of the foreign key column in the child table.
+        metadata (dict):
+            The metadata.
+
+    Returns:
+        plotly.graph_objects._figure.Figure
+    """
+    parent_primary_key = None
+    for relation_dict in metadata.get('relationships', []):
```

Comment:
minor: I don't think we need the dict suffix

Reply:
Yes, done in [76fcf86](https://github.com/sdv-dev/SDMetrics/pull/384/commits/76fcf862286df85acac9ee2977216ac5c72f5a30)